src/utils/url.ts

```diff
@@ -27,6 +32,44 @@ export const transactionIdAndSubmissionIdExistInRequest = (req: Request): boolea
 const getTransactionIdFromRequestParams = (req: Request): string => req.params[config.ROUTE_PARAM_TRANSACTION_ID];
 const getSubmissionIdFromRequestParams = (req: Request): string => req.params[config.ROUTE_PARAM_SUBMISSION_ID];
 
+/**
+ * This is required to extract the transactionId and submissionId prior to the request reaching its intended mount path
+ */
+export const getTransactionIdAndSubmissionIdFromOriginalUrl = (req: Request): TransactionIdAndSubmissionId | undefined => {
+  try {
+    logger.infoRequest(req, `extracting transactionId and submissionId from the req.originalUrl: ${req.originalUrl}`);
+    if (!req.originalUrl.includes('/transaction/') || !req.originalUrl.includes('/submission/')) {
+      return;
+    }
+    const elements = req.originalUrl.replace(/\/{2,}/g, "/").split("/");
+    const transactionIndex = elements.indexOf("transaction") + 1;
+    const submissionIndex = elements.indexOf("submission") + 1;
+    if (!elements[transactionIndex] || !elements[submissionIndex]) {
+      return;
+    }
+    return {
+      transactionId: elements[transactionIndex],
+      submissionId: elements[submissionIndex],
+    };
+  } catch (e) {
+    logger.infoRequest(req, `error extracting transactionId and submissionId from req.originalUrl: ${req.originalUrl} with error object: ${e}`);
```

Comment:
what happens to the execution if an error is caught, it doesn't seem to return or throw and I have a feeling this might cause the web journey to hang. Could we re-throw the error (or not catch it) so the errorHandler controller could catch it and display the 'service unavailable page'?

Reply:
Also, should logger.infoRequest be logger.errorRequest if it is a non recoverable error

Reply:
In the unlikely event that this happens, we catch, log, and return `undefined` so that the calling method uses the fallback baseUrl.. no need to re-throw as we need to set the fallback baseUrl and proceed to `next`..

Yes, `logger.infoRequest` should be `logger.errorRequest`. This has been updated. Thanks for the good spot.